plugins/scaffolder-backend/src/scaffolder/stages/prepare/github.ts

```diff
@@ -64,7 +64,22 @@ export class GithubPreparer implements PreparerBase {
         }
       : {};
 
-    await Clone.clone(repositoryCheckoutUrl, tempDir, cloneOptions);
+    const repository = await Clone.clone(
+      repositoryCheckoutUrl,
+      tempDir,
+      cloneOptions,
```

Comment:
Could we just provide `checkoutBranch` here instead? I imagine the behavior of not switching the branch if it's not found is mainly just confusing, or is there a good reason for it to behave that way?

Reply:
Yes, good call. I never used gitnode before so I'm not familiar with the API. Thanks